src/components/main/main.tsx

```diff
@@ -4,16 +4,57 @@ import { Home } from 'pages/pageHome/pageHome';
 import { NotFound } from 'pages/pageNotFound/pageNotFound';
 import React from 'react';
 import { Routes, Route } from 'react-router-dom';
+import { AddNewCat } from 'pages/pageHome/modal/addNewCat';
+import data from '../../storage/data.json';
+
+export type Cat = {
+  id: string;
+  length: string;
+  image?: string;
+  minWeight: number;
+  maxWeight: number;
+  minLife: number;
+  maxLife: number;
+  name: string;
+};
+
+export type Cats = {
+  cats: Cat[];
+};
 
 export class Main extends React.Component {
+  state = { cats: data, isActive: false };
+
+  createCat = (cat: Cat) => {
+    this.setState({ cats: [...this.state.cats, cat] });
+    alert('New cat created');
+  };
+
+  toggleModal = () => {
+    this.setState({ isActive: this.state.isActive ? false : true });
```

Comment:
```suggestion
    this.setState({ isActive: !this.state.isActive });
```

а вообще нужно делать через калбек, тк. ты используешь предыдущий стейт